suite-common/wallet-core/src/accounts/accountsReducer.ts

```diff
@@ -86,15 +87,35 @@ export const prepareAccountsReducer = createReducerWithExtraDeps(
 
 export const selectAccounts = (state: AccountsRootState) => state.wallet.accounts;
 
-export const selectAccount = (state: AccountsRootState, accountKey: string) =>
-    state.wallet.accounts.find(account => account.key === accountKey);
+export const selectAccount = createSelector(
+    selectAccounts,
+    (_state: AccountsRootState, accountKey: string) => accountKey,
+    (accounts, accountKey) => accounts.find(account => account.key === accountKey),
+);
+
+export const selectDeviceNetworkAccounts = createSelector(
+    selectAccounts,
+    (_state: AccountsRootState, deviceState: string, networkSymbol: NetworkSymbol) => ({
+        deviceState,
+        networkSymbol,
+    }),
+    (accounts, params) =>
+        accounts.filter(
+            account =>
+                account.deviceState === params.deviceState &&
+                account.symbol === params.networkSymbol,
+        ),
+);
 
-// So far, mobile app doesn't persis data between app runs so until this is resolved
-// account names are just "Bitcon #<ACCOUNT_INDEX>"
-export const selectAccountName = createSelector(
+export const selectAccountLabel = createSelector(
     [selectAccount, selectAccounts],
     (account, accounts) => {
-        const accountIndex = accounts.findIndex(acc => acc.descriptor === account?.descriptor);
-        return `Bitcoin #${accountIndex + 1}`;
+        const accountData = accounts.find(acc => acc.descriptor === account?.descriptor);
+        if (accountData) {
```

Comment:
Wouldn't be better to return for example empty string instead of undefined? IDK

Reply:
I think it does not matter. In the end, this case should not occur at all... it is guaranteed by the discovery in suite (you always have an account with name - e.g. Bitcoin #XY or your custom label if you use Dropbox integration there...) and it will be guaranteed in suite native too after we add validation to prevent empty string there.